api.ts

```diff
@@ -1660,15 +1683,15 @@ export interface BaseFormulaDef<ParamDefsT extends ParamDefs, ResultT extends st
 export type StringFormulaDef<ParamDefsT extends ParamDefs> = BaseFormulaDef<ParamDefsT, string> & {
   resultType: ValueType.String;
   execute(params: ParamValues<ParamDefsT>, context: ExecutionContext): Promise<string> | string;
-} & ({schema?: StringSchema} | {codaType?: StringHintTypes});
+} & {schema?: StringSchema; codaType?: StringHintTypes};
```

Comment:
How come these are changing? I think these were intentionally mutually-exclusive before.

Reply:
I'm not sure if I'm not writing this well or if TS is not able to recognize something that a human is able to recognize. When I use FormulaOptions<ParamDefsT, StringFormulaDef<...>>, TS makes codaType and schema unknown even if we check 'codaType' in fullDefinition. I've reverted this now and have added some types to cast to that make TS happy.